apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -150,111 +38,82 @@ suite('Contacts', function(done) {
     window.fb = mockReaderUtils;
   });
 
-  teardown(function() {
-    navigator.mozContacts.mHistory.length = 0;
-  });
-
   suiteTeardown(function() {
     navigator.mozContacts = nativeMozContacts;
     window.fb = realFb;
+    //Restoring the native API
+    navigator.mozContacts.find.restore();
   });
 
   suite('Contacts.findContactByString, single-term', function() {
 
     test('(string[tel,givenName,familyName], ...) Match', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('O\'Hare').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
```

Comment:
Because we pass filterValue as the arguement for contacts.findby() functions but filter object as arguement for mozContacts.find(), we need to use sinon.match with every withArgs.